test/e2e/apibinding/apibinding_test.go

```diff
@@ -0,0 +1,217 @@
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "error waiting for source workspace to be ready")
+
+	server.Artifact(t, func() (runtime.Object, error) {
+		return orgKcpClient.TenancyV1alpha1().ClusterWorkspaces().Get(ctx, sourceWorkspace.Name, metav1.GetOptions{})
+	})
+
+	t.Logf("Creating target workspace in org lcluster %s", orgClusterName)
+	targetWorkspace := &tenancyv1alpha1.ClusterWorkspace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "target-",
+		},
+	}
+	targetWorkspace, err = orgKcpClient.TenancyV1alpha1().ClusterWorkspaces().Create(ctx, targetWorkspace, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating target workspace")
+	require.Eventually(t, func() bool {
+		ws, err := orgKcpClient.TenancyV1alpha1().ClusterWorkspaces().Get(ctx, targetWorkspace.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error getting target workspace")
+		return ws.Status.Phase == tenancyv1alpha1.ClusterWorkspacePhaseReady
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "error waiting for target workspace to be ready")
+
+	server.Artifact(t, func() (runtime.Object, error) {
+		return orgKcpClient.TenancyV1alpha1().ClusterWorkspaces().Get(ctx, targetWorkspace.Name, metav1.GetOptions{})
+	})
+
+	var (
+		sourceWorkspaceClusterName = org + ":" + sourceWorkspace.Name
+		targetWorkspaceClusterName = org + ":" + targetWorkspace.Name
+	)
+
+	t.Logf("Install a cowboys APIResourceSchema into workspace %q", sourceWorkspace.Name)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(kcpClients.Cluster(sourceWorkspaceClusterName).Discovery()))
+	err = helpers.CreateResourceFromFS(ctx, dynamicClients.Cluster(sourceWorkspaceClusterName), mapper, "apiresourceschema_cowboys.yaml", testFiles)
+	require.NoError(t, err)
+
+	t.Logf("Create an APIExport for it")
+	cowboysAPIExport := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "today-cowboys",
+		},
+		Spec: apisv1alpha1.APIExportSpec{
+			LatestResourceSchemas: []string{"today.cowboys.wildwest.dev"},
+		},
+	}
+	_, err = kcpClients.Cluster(sourceWorkspaceClusterName).ApisV1alpha1().APIExports().Create(ctx, cowboysAPIExport, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Create an APIBinding in workspace %q that points to the today-cowboys export", targetWorkspace.Name)
+	apiBinding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cowboys",
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					WorkspaceName: sourceWorkspace.Name,
+					ExportName:    cowboysAPIExport.Name,
+				},
+			},
+		},
+	}
+
+	_, err = kcpClients.Cluster(targetWorkspaceClusterName).ApisV1alpha1().APIBindings().Create(ctx, apiBinding, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Make sure %s API group does NOT show up in workspace %q group discovery", wildwest.GroupName, sourceWorkspace.Name)
```

Comment:
(No action required) Aha, so exporting a type doesn't require it to be installed in a workspace? And an apiresourceschema is basically a crd that isn't reconciled into an api without an apibinding?

Reply:
A APIResourceSchema is like a snapshot (in time) of a CRD, but yes: it is not served in that workspace. The objects are in the consumer workspaces.